File: week-04/day-3/Exercise_4/count_letters.py
# Count Letters

# Write a function, that takes a string as an argument and returns a 
# dictionary with all letters in the string as keys, and numbers as 
# values that shows how many occurrences there are.
# Create a test for that.
class Letters:
    def __init__(self,string="abcdefghijkabcsdklvknsccc",dictionary={}):
        self.dictionary = dictionary
        self.string = string
    
    def letter_counter(self):
        for i in self.string:
            self.dictionary[i] = self.string.count(i)
        return self.dictionary

File: week-09/day-1/Exercises/exercise5/countletters.js
'use strict'
// Write a function, that takes a string as an argument and returns a dictionary with all letters in 
// the string as keys, and numbers as values that shows how many occurrences there are.
// Create a test for that.

let CountLetters = function(str) {
    try {
        if(typeof str === 'string') {
            let dictionary = {};
            let strArray = str.split('');
            for(let i = 0; i < strArray.length; i++) {
                let filtered = strArray.filter(function(el) {
                    return el === strArray[i]
                })
                dictionary[strArray[i]] = filtered.length;
            }
            return dictionary;
        } else {
            throw new Error('Not a string');
        }
    } catch (error) {
        return error.message;
    }
}

module.exports = CountLetters;